deluge/core/torrent.py

```diff
@@ -1024,13 +1026,35 @@ def get_status(self, keys, diff=False, update=False, all_keys=False):
 
         return status_dict
 
-    def update_status(self, status):
+    def update_status(self) -> 'lt.torrent_status':
+        """Get the torrent status fresh, not from cache.
+
+        This should be used when a guaranteed fresh status is needed rather than
+        `torrent.handle.status()` because it will update the cache as well.
+        """
+        self.status = self.handle.status()
+        return self.status
+
+    @property
+    def status(self) -> 'lt.torrent_status':
```

Comment:
I'm not sure if it would make it more or less complex to use `cached_property` but moving to a status property looks good :+1:

Reply:
Took a peek at cached_property, doesn't look like it helps us much.